src/geventhttpclient/connectionpool.py

```diff
@@ -196,6 +196,8 @@ def __init__(self, host, port, **kw):
 
         def after_connect(self, sock):
             super(SSLConnectionPool, self).after_connect(sock)
+            sock.check_hostname = True
```

Comment:
It feels like this shouldn't be set true in insecure mode.
Does it remove the need for `match_hostname` below ?